From 1bc387f8feaab9020be72e88cf26ccc1a67a6a10 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 22 Aug 2020 23:33:48 +0200
Subject: [PATCH] meson: allow the user to disable opencv

Allow the user to really disable opencv through meson (i.e.
-Dopencv=disabled).

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status:
https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1533]
---
 gst-libs/gst/opencv/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst-libs/gst/opencv/meson.build b/gst-libs/gst/opencv/meson.build
index 6cc4602f3..fb6c46e40 100644
--- a/gst-libs/gst/opencv/meson.build
+++ b/gst-libs/gst/opencv/meson.build
@@ -13,7 +13,7 @@ opencv_dep = dependency('opencv', version : '>= 3.0.0', required : false)
 if not opencv_dep.found()
   opencv_dep = dependency('opencv4', version : '>= 4.0.0', required : false)
 endif
-if opencv_dep.found()
+if not get_option('opencv').disabled() and opencv_dep.found()
   gstopencv = library('gstopencv-' + api_version,
     opencv_sources,
     c_args : gst_plugins_bad_args + ['-DBUILDING_GST_OPENCV'],
-- 
2.28.0

